tools/codeowners-linter/ci.yml

```diff
@@ -0,0 +1,26 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
```

Comment:
Why do we need a new ci? Is there a reason we don't include this under the existing codeowners tools directory? They seem like they are connected enough to be together.

Reply:
@weshaggard because this is a separate tool that's being built and uploaded to the dev feed. The codeowners ci.yml isn't for the codeowners library but for the retrieve-codeowners tool. I'd like to keep things separate.

Reply:
OK once we combine the libraries we should probably put all the codeowners tools under the same directory and have the one pipeline. It is OK if we release more than one library or tool from the same pipeline.